kuma/javascript/src/newsletter.jsx

```diff
@@ -89,76 +91,71 @@ export default function Newsletter() {
     const submit = event => {
         let newsletterForm = newsletterFormRef.current;
 
-        // if skipFetch
-        if (newsletterForm) {
-            if (newsletterForm.dataset['skipFetch']) {
-                /*
-                 * An error occured while attempting to subscribe
-                 * the user via Ajax, but no specific error was provided. We
-                 * therefore just send the user directly to the Mozorg
-                 * newsletter subscription page and log the occurence
-                 */
-                ga('send', {
-                    hitType: 'event',
-                    eventCategory: 'newsletter',
-                    eventAction: 'progression',
-                    eventLabel: 'error-forward'
-                });
-                return true;
-            }
-
-            event.preventDefault();
+        if (!newsletterForm) {
+            return;
+        }
 
-            let params = new URLSearchParams(
-                new FormData(newsletterForm)
-            ).toString();
+        // if skipFetch
+        if (newsletterForm.dataset['skipFetch']) {
+            /*
+             * An error occured while attempting to subscribe
+             * the user via Ajax, but no specific error was provided. We
+             * therefore just send the user directly to the Mozorg
+             * newsletter subscription page and log the occurence
+             */
+            ga('send', {
+                hitType: 'event',
+                eventCategory: 'newsletter',
+                eventAction: 'progression',
+                eventLabel: 'error-forward'
+            });
+            return true;
+        }
 
-            fetch(newsletterSubscribeURL, {
-                method: 'POST',
-                headers: {
-                    'X-Requested-With': 'XMLHttpRequest',
-                    'Content-type': 'application/x-www-form-urlencoded'
-                },
-                body: params
-            }).then(response => {
-                response.json().then(responseJSON => {
-                    if (responseJSON.success !== 'success') {
-                        let errorsArray = responseJSON.errors;
+        event.preventDefault();
+        let params = new URLSearchParams(
+            new FormData(newsletterForm)
+        ).toString();
+        fetch(newsletterSubscribeURL, {
+            method: 'POST',
+            headers: {
+                'X-Requested-With': 'XMLHttpRequest',
+                'Content-type': 'application/x-www-form-urlencoded'
+            },
+            body: params
+        })
+            .then(response => response.json())
+            .then(({ success, errors }) => {
+                if (success === 'success') {
+                    return;
+                }
```

Comment:
It looks like this is the piece of code that will run if the subscription completed successfully. If so, there are a couple of things we need to do. Essentially everything from https://github.com/mozilla/kuma/pull/5772/files#diff-92ad4e492a736caacb5293381494f20eR142 - 150 needs to be moved inside this conditional.

At the moment, once the `fetch` completes successfully, the newsletter is hidden, but no message is shown to the user.

Reply:
Thanks, indeed. Apparently I only read the code with half a mind. I made those changes.

Also I noticed that `response.success` is a boolean (which imo also makes more sense), so that block would've never run. Or is there a case in which `response.success` would be the `success` string?

Reply:
Nope, not from what I can see.